dashboard/common/db_util.py

```diff
@@ -131,4 +135,47 @@ def get_canvas_data_date():
                 return date
     except Exception:
         logger.info("Value could not be found from metadata", exc_info = True)
-    return datetime.min
\ No newline at end of file
+    return datetime.min
+
+class CourseEnrollment:
+
+    def __init__(self, course_name:str, course_id:int, course_description:str, enrollment_type:str, current_grade:float, final_grade:float) -> None:
+        self.course_name = course_name
+        self.course_id = course_id
+        self.course_description = course_description
+        self.enrollment_type = enrollment_type
+        self.current_grade = current_grade
+        self.final_grade = final_grade
+
+    def __str__(self):
+        result = "{'course_name':'"+self.course_name+"','course_id':'"+str(self.course_id)+"','course_description':'"+str(self.course_description)+"','enrollment_type':'"+self.enrollment_type+"','current_grade':'"+str(self.current_grade)+"','final_grade':'"+str(self.final_grade)+"'}"
+        return result
+
+class CourseEnrollmentEncoder(JSONEncoder):
+    def default(self,o:CourseEnrollment):
+        return o.__dict__
+
+
+
+def get_course_enrollment_info(username) -> List[CourseEnrollment]:
+    logger.info(get_course_enrollment_info.__name__)
+    courses : []
+    course_enrollment : List[CourseEnrollment] = []
+    try:
+        with django.db.connection.cursor() as cursor:
+            cursor.execute("select c.name, c.id, u.enrollment_type, u.current_grade, u.final_grade FROM student_dashboard.`user` u, student_dashboard.course c where u.sis_name=%s and u.course_id =c.id", [username]);
+            courses = cursor.fetchall()
+            if courses is not None:
+                for course in courses:
+                    name=course[0]
+                    course_id=incremented_id_to_canvas_id(course[1])
+                    enrollment_type=course[2]
+                    current_grade = course[3]
+                    final_grade = course[4]
+                    enrollment = CourseEnrollment(name, course_id, None, enrollment_type, current_grade, final_grade)
+                    logger.info("APPENDING "+str(enrollment))
+                    course_enrollment.append(enrollment)
+    except Exception as err:
+        logger.info("Error in get_course_enrollment_info: "+str(err), exc_info = True)
+    return course_enrollment
+
```

Comment:
I'd like to have a discussion with you about why all this is necessary when we already have an existing function, `get_user_courses_info` that could be expanded to include `enrollment_type`. In an ideal world, we'd change the database so there was an `enrollment` table and model, which would make our lives easier, but that can be a future effort. It feels like, while your intention was good, it's duplicating a lot of code and introducing things that may not be used.

Reply:
Was trying to make headway towards using type hinting and being new i didn't want to create any unexpected side effects.  I'm fine with doing it differently if there is an advantage.

Reply:
Done